cumulus/client/consensus/common/src/lib.rs

```diff
@@ -389,19 +484,19 @@ pub async fn find_potential_parents<B: BlockT>(
 		}
 
 		// push children onto search frontier.
-		for child in client.blockchain().children(hash).ok().into_iter().flatten() {
+		for child in backend.blockchain().children(hash).ok().into_iter().flatten() {
```

Comment:
We could consider iterating from the leaves along the parents here. (not in this PR)